src/components/workspace/collabEditing/helper.ts

```diff
@@ -1,40 +1,79 @@
-import { LINKS } from '../../../utils/constants';
-
-export function checkSessionIdExists(
-  sessionId: string,
-  onSessionIdExists: () => void,
-  onSessionIdNotExist: () => void,
-  onServerUnreachable: () => void
-) {
-  const xmlhttp = new XMLHttpRequest();
-  xmlhttp.onreadystatechange = () => {
-    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
-      // Successfully reached server to verify ID
-      const state = JSON.parse(xmlhttp.responseText).state;
-      if (state === true) {
-        // Session ID exists
-        onSessionIdExists();
-      } else {
-        onSessionIdNotExist();
-      }
-    } else if (xmlhttp.readyState === 4 && xmlhttp.status !== 200) {
-      // Cannot reach server
-      onServerUnreachable();
-    }
-  };
-
-  xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/' + sessionId, true);
-  xmlhttp.send();
-}
-
-export function createNewSession(onSessionCreated: (sessionId: string) => void) {
-  const xmlhttp = new XMLHttpRequest();
-  xmlhttp.onreadystatechange = () => {
-    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
-      const id = JSON.parse(xmlhttp.responseText).id;
-      onSessionCreated(id);
-    }
-  };
-  xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/latest/', true);
-  xmlhttp.send();
-}
+import { LINKS } from '../../../utils/constants';
+
+enum XMLHttpReadyState {
+  UNSENT = 0,
+  OPENED = 1,
+  HEADERS_RECEIVED = 2,
+  LOADING = 3,
+  DONE = 4
+}
+
+enum XMLHttpStatus {
+  OK = 200,
+  PAGE_NOT_FOUND = 404
+}
+
+export function checkSessionIdExists(
+  sessionId: string,
+  onSessionIdExists: () => void,
+  onSessionIdNotExist: () => void,
+  onServerUnreachable: () => void
+) {
+  const xmlhttp = new XMLHttpRequest();
```

Comment:
Add types here! 

`const xmlhttp : XMLHttpRequest = new XMLHttpRequest();`

(Do the same for the rest)

Reply:
I'm not sure why is there a need to put types here? Isn't the type inference from typescript sufficient?

Reply:
Yes your right! It may just be better to use type inference here in this case. I was reviewing by purely looking at your code on Github so missed this detail!